lib/Sema/TypeCheckType.cpp

```diff
@@ -2225,12 +2225,21 @@ Type TypeResolver::resolveAttributedType(TypeAttributes &attrs,
               SourceLoc(), Context.getIdentifier(protocolName));
         }
       }
+      
+      DifferentiabilityKind diffkind = DifferentiabilityKind::NonDifferentiable;
+      if (attrs.has(TAK_differentiable)) {
+        if (attrs.linear) {
+          diffkind = DifferentiabilityKind::Linear;
```

Comment:
Use a ternary here?

Reply:
Ahh yes I simplified my nested ternary too much! Before I had the double nested ternary, but now I can just have one.